lib/swagger.js

```diff
@@ -1549,29 +1546,18 @@ SwaggerAuthorizations.prototype.remove = function(name) {
 };
 
 SwaggerAuthorizations.prototype.apply = function(obj, authorizations) {
-  var status = null;
-  var key;
-
-  // if the "authorizations" key is undefined, or has an empty array, add all keys
-  if(typeof authorizations === 'undefined' || Object.keys(authorizations).length == 0) {
-    for (key in this.authz) {
-      value = this.authz[key];
-      result = value.apply(obj, authorizations);
-      if (result === true)
-        status = true;
-    }
-  }
-  else {
-    for(name in authorizations) {
-      for (key in this.authz) {
-        if(key == name) {
-          value = this.authz[key];
-          result = value.apply(obj, authorizations);
-          if (result === true)
-            status = true;
-        }
-      }      
-    }
+  var status, key, value, result;
+  status = true;
+
+  for(name in authorizations) {
+    for(key in this.authz) {
+      if(key == name) {
+        value = this.authz[key];
+        result = value.apply(obj, authorizations);
+        if (result === false)
+          status = false;
+      }
+    }      
```

Comment:
This change will introduce a different behavior, namely, _all_ authorizations should be applied if the object does not specify an authorization requirement (line 1556).  If we remove lines 1556 to 1563, we will lose this logic, and only apply authorizations where they're matching by name